### PR TITLE
Request to merge feature to development

### PR DESCRIPTION
**Pull request by kimothokamau**
This is a request to merge changes made in the feature branch to the development branch. 
The features implemented allow:
- Members to signup using email, username, name and password
- Authenticated members to make posts and view other posts
- Non-authenticated members to view posts but not the author of the posts

### CODE REVIEW

**Review by rindrajosia — APPROVED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- User can sign in :+1: 
- Signed-in user can create new post :100: 
- Signed-in user can see post's author :1st_place_medal: 
- User can sign_out :100: 
- It displays Flash messages :+1: 

To improve :factory_worker: [REQUIRED]:

- [x] In your APP guest can see both  post's author and post's text however, guest must be able to see only post's text([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- [x] It will be perfect if you remove logic in your view and partial files and put it in your helpers. It is much easier to read code that is lean and skinny, as well as serving a single purpose.
- [x] Posts#routes: It creates the application routes using resources :posts, only: [:new, :create, :index] ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- [x] PostsController#new: it initializes an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- [ ] It will be perfect if you add screenshot in your readme file.
- [ ] In the `Getting started section` you should ask user to create DB via `rails db:create` and to migrate it via `rails db:migrate` before `rails server`
- [ ] Please, make sure that` issues page` link redirect to the correct page.

Nice to have:

- [ ] 

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 
Nice work making the requested changes. 
I was not able to run the app locally. Let's make some changes to the PR.
- [ ] I got this error when running locally
```ruby
Webpacker can't find application.js in /home/afani/members-only/public/packs/manifest.json. Possible causes:
1. You want to set webpacker.yml value of compile to true for your environment
   unless you are using the `webpack -w` or the webpack-dev-server.
2. webpack has not yet re-run to reflect updates.
3. You have misconfigured Webpacker's config/webpacker.yml file.
```
- [ ] Add this to the `routes.rb`: `resources :posts, only: [:new, :create, :index] `


### Martin Afani
#### Slack: whiz25

**Comment by mimipeshy** [comment]
## STATUS: CHANGES REQUESTED 🔴 ♻️ 
Hello, team @kimothokamau 👋🏾 👋🏾 this is great work you have put in this milestone. You have worked on the TSE comments left and that is commendable 😄 
However, before approval kindly work on the following  comments I will leave below⬇️ 

- [ ]  Kindly remove logic from your views. Logic includes **loops and if statements**. This will ensure your project follows the MVC pattern. You can create all your logic inside the [helpers](http://codefol.io/posts/Where-Do-I-Put-My-Code/
). 

For some reason, I cannot comment directly on your code where the logic is which is due to the few commits you have made but I will make a list for you here ⬇️ 

- app/views/posts/index.html.erb
- app/views/posts/_feed.html.erb
- /app/views/posts/_form.html.erb

- [ ] Kindly add more commit messages in your PR 

That's all 😄, kindly submit another review when done
If you need any clarification, kindly post your comment here and tag me @mimipeshy and will be glad to help

Regards,
Peris Ndanu

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,5 @@
 Rails.application.routes.draw do
-  devise_for :users, :controllers => { registrations: 'registrations'}
   resources :posts
```
Comment (mimipeshy) [inline config/routes.rb:2]: - [ ] Restrict routes to  resources :posts, only: [:new, :create, :index]
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

**Review by BrianSammit — APPROVED** [review]
STATUS APPROVED :green_circle: :clap: 
----------
Hello @kimothokamau You did a great project so far. :muscle: Good job working on the required changes :100: :+1: 
---------
Now you can merge this branch. :clap: :smile: 
HAPPY CODING!! AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

Well-Done-Applause

**Pull request merged**